apps/condo/domains/meter/components/CreateMeterReadingsActionBar.tsx

```diff
@@ -57,7 +57,12 @@ export const CreateMeterReadingsActionBar = ({
                     }
 
                     const hasWrongNumberOfValues = Object.keys(newMeterReadings).reduce((result, meterId) => {
-                        return result || Object.keys(get(newMeterReadings, meterId, {})).length !== get(metersByIds, [meterId, 'numberOfTariffs'])
+                        const meterReadings = get(newMeterReadings, meterId, {})
+                        const numberOfValues = Object.keys(meterReadings).filter(key => key !== 'date').length
```

Comment:
What about calculating explicitly fields with values instead of implicit calculation by excluding other fields?

Reply:
yes, in the future we might want to add some other fields, although right now i don't know which ones. but to make it future-proof i'll explicitly go through values only